woocommerce/woocommerce-ios
# Add WooCommerce installation with web view

Closes: #7236 


### Description

This PR fixes the incorrect view displayed when a user tries to log in to a site without WooCommerce. Here, we're displaying a new screen for the error with a button to install Woo from the app.

The Woo Setup screen is using the same `PluginSetupWebViewController` from #7395. Here for the Woo Setup flow, we don't have a real redirect URL for the completion, so I have to listen to the changes in the web view URL  to detect the end of the flow.

Edge case: if a site is non-atomic, it bypasses the check for Jetpack and the app will show the No Woo error screen. In this case, if the user taps the Install WooCommerce button, they will be presented with a button to upgrade their account, which we don't want. So for now, I'm hiding the install button for this case.

The design and copies for this screen is not final yet since we haven't got them reviewed, but we can update them in another PR.

### Testing instructions

**Case 1**: Self-hosted store with Jetpack and without WooCommerce.
1. Select Enter site address and enter your address.
2. Proceed to log in with either WP credentials or site credentials.
3. After login succeeds, notice that the app navigates to the No Woo error screen. In Xcode console, notice "🔵 Tracked login_woocommerce_error_shown".
4. If your account has other connected stores, notice that the link button for "See connected stores" is visible. Tapping it should navigate to the site picker with the sites that have WooCommerce.  If your account doesn't have other connected stores, notice that the link button is not visible.
5. Select Install WooCommerce. Notice that the app navigates to a web view for installing Woo. You shouldn't have to re-enter WP.com credentials here.  In Xcode console, notice "🔵 Tracked login_woocommerce_setup_button_tapped".
6. Dismiss the view, notice "🔵 Tracked login_woocommerce_setup_dismissed, properties: [AnyHashable("source"): "web"]" in Xcode console.
7. Select the Install WooCommerce button again. 
9. Select "Install & Activate WooCommerce" button on the web page. When the setup completes, notice that a loading screen is displayed and then the app navigates to the home screen. In Xcode console, notice "🔵 Tracked login_woocommerce_setup_completed".

**Case 2**: Non-atomic site
1. Select Enter site address and enter your site address.
2. Proceed to log in with either WP credentials.
3. After login succeeds, notice that the app navigates to the No Woo error screen.
4. If your account has other connected stores, notice that the link button for "See connected stores" is visible. Tapping it should navigate to the site picker with the sites that have WooCommerce. Otherwise, the button should be hidden.
5. Notice that the Install WooCommerce button should be hidden in this case.

### Screenshots


Install flow for a self-hosted site:

https://user-images.githubusercontent.com/5533851/182523399-d2a89e7a-c902-438f-b3e5-21e190d4f610.mp4

No Woo error screen for a non-atomic site without connected stores:

<img src="https://user-images.githubusercontent.com/5533851/182648422-0b2243d4-2431-4ca6-a37e-9dc2baceef13.png" width=320 />


---
